src/xml.tsx

```diff
@@ -435,6 +438,40 @@ export function parse(source: string, middleware?: Middleware): JsxAST | null {
     return name;
   }
 
+  function collectCssVariables(source: string) {
+    const styleRegex = /<style[^>]*>([\s\S]*?)<\/style>/gi;
+    let match;
+
+    while ((match = styleRegex.exec(source)) !== null) {
+      const styleContent = match[1];
+      const variableRegex = /--([^:]+):\s*([^;]+);/g;
+      let varMatch;
+      while ((varMatch = variableRegex.exec(styleContent)) !== null) {
+        const [, name, value] = varMatch;
+        cssVariables[`--${name.trim()}`] = value.trim();
+      }
+    }
+  }
+
+  function parseStyleAttribute(styleString: string): Styles {
+    const style: Styles = {};
+    const declarations = styleString.split(';').filter((v) => v.trim());
+
+    for (const declaration of declarations) {
+      const [property, value] = declaration.split(':').map((s) => s.trim());
+      if (property && value) {
+        const camelProperty = camelCase(property);
+        if (value.startsWith('var(')) {
+          const variableName = value.slice(4, -1).trim();
+          style[camelProperty] = cssVariables[variableName] || value;
+        } else {
+          style[camelProperty] = value;
+        }
+      }
+    }
+    return style;
+  }
+
```

Comment:
I don't think that it should be done here, as `SvgXml` does not support CSS styles at all.

Reply:
Good point. handle that in `css.tsx` file.